src/stores/installed-apps-slice.ts

```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export interface InstalledApps {
+	vscode: boolean;
+	phpstorm: boolean;
+	webstorm: boolean;
+	windsurf: boolean;
+	cursor: boolean;
+	iterm: boolean;
+}
+
+const initialState: InstalledApps = {
+	vscode: false,
+	phpstorm: false,
+	webstorm: false,
+	windsurf: false,
+	cursor: false,
+	iterm: false,
+};
+
+export const fetchInstalledApps = createAsyncThunk(
+	'installedApps/fetchInstalledApps',
+	async () => {
+		const installedApps = await getIpcApi().getInstalledApps();
+		return installedApps;
+	}
+);
+
+const installedAppsSlice = createSlice( {
+	name: 'installedApps',
+	initialState,
+	reducers: {},
+	extraReducers: ( builder ) => {
+		builder.addCase( fetchInstalledApps.fulfilled, ( state, action ) => {
+			// Update the state with the fetched installed apps
+			Object.assign( state, action.payload );
+		} );
+	},
+} );
+
+export const installedAppsActions = installedAppsSlice.actions;
+export const reducer = installedAppsSlice.reducer;
+export const selectInstalledApps = ( state: { installedApps: InstalledApps } ) =>
+	state.installedApps;
```

Comment:
Given how simple this selector is, I don't really think it's needed. For future reference, `createSlice` accepts a `selectors` object where each selector automatically receives just the local state slice.

Reply:
Thanks, I didn't know that we have that option for `createSlice`, that's pretty handy! 👍

Reply:
`createSlice` is honestly a dream to work with compared to "classic" Redux 😅